src/streaming/mod.rs

```diff
@@ -265,6 +270,53 @@ mod request {
             Self::AbortChunkedWrite(request)
         }
     }
+
+    #[derive(Debug, PartialEq, Eq, Deserialize, Serialize)]
+    pub struct PartialReadFile {
+        pub location: Location,
+        pub path: PathBuf,
+        pub offset: usize,
+        pub length: usize,
+    }
```

Comment:
Maybe we should drop the length to make it more transparent.  The caller can still `truncate` if needed.

Reply:
I'm in favour of keeping it. The NK3 is already pretty slow (at least for the opcard app).